src/cryptography/x509.py

```diff
@@ -460,6 +462,98 @@ def __ne__(self, other):
     access_location = utils.read_only_property("_access_location")
 
 
+class CertificatePolicies(object):
+    def __init__(self, policies):
+        if not all(isinstance(x, PolicyInformation) for x in policies):
+            raise TypeError(
+                "Every item in the policies list must be a "
+                "PolicyInformation"
+            )
+
+        self._policies = policies
+
+    def __iter__(self):
+        return iter(self._policies)
+
+    def __len__(self):
+        return len(self._policies)
+
+    def __repr__(self):
+        return "<CertificatePolicies({0})>".format(self._policies)
+
+
+class PolicyInformation(object):
+    def __init__(self, policy_identifier, policy_qualifiers):
+        if not isinstance(policy_identifier, ObjectIdentifier):
+            raise TypeError("policy_identifier must be an ObjectIdentifier")
+
+        self._policy_identifier = policy_identifier
+        if policy_qualifiers and not all(
```

Comment:
Oh, bleh, right.

On Mon, May 11, 2015 at 11:45 AM, Paul Kehrer <notifications@github.com>
wrote:

> In src/cryptography/x509.py
> <https://github.com/pyca/cryptography/pull/1853#discussion_r30050815>:
>
> > +        return iter(self._policies)
> > +
> > +    def __len__(self):
> > +        return len(self._policies)
> > +
> > +    def __repr__(self):
> > +        return "<CertificatePolicies({0})>".format(self._policies)
> > +
> > +
> > +class PolicyInformation(object):
> > +    def __init__(self, policy_identifier, policy_qualifiers):
> > +        if not isinstance(policy_identifier, ObjectIdentifier):
> > +            raise TypeError("policy_identifier must be an ObjectIdentifier")
> > +
> > +        self._policy_identifier = policy_identifier
> > +        if policy_qualifiers and not all(
>
> Wouldn't this fail on None because NoneType is not iterable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pyca/cryptography/pull/1853/files#r30050815>.
>



-- 
"I disapprove of what you say, but I will defend to the death your right to
say it." -- Evelyn Beatrice Hall (summarizing Voltaire)
"The people's good is the highest law." -- Cicero
GPG Key fingerprint: 125F 5C67 DFE9 4084